root/components/Header.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+const React = require('react');
+const {l} = require('../static/scripts/common/i18n');
+const EntityTabs = require('./EntityTabs');
+const Frag = require('./Frag');
+const SubHeader = require('./SubHeader');
+const EntityLink = require('../static/scripts/common/components/EntityLink');
+const TaggerIcon = require('../static/scripts/common/components/TaggerIcon');
+
+type Props = {
+  entity: EntityT;
+  editTab?: string | React.Node;
+  headerClass: string;
+  heading?: string | React.Node;
+  page: string;
+  subHeading: string;
+};
+
+const Header = ({entity, headerClass, heading, subHeading, ...props}: Props) => (
+  <Frag>
+    <div className={headerClass}>
+      {$c.session && $c.session.tport ? <TaggerIcon entity={entity} /> : null}
```

Comment:
Hmm, it's not clear why the `tport` is checked outside the component, but the entity types are checked inside the component. Maybe just move the entity checks here too? But in either case, move the entity array constant (`['recording', 'release']`) outside the render definition so that it isn't reallocated. (We actually have a babel plugin enabled that does [similar magic for constant react elements](https://babeljs.io/docs/plugins/transform-react-constant-elements/).)